functions/pubsub/handlers.js

```diff
@@ -0,0 +1,271 @@
+        creatorIsCivicLiker: isSubscribedCivicLiker,
+        followerDisplayName: email,
+        nftTitle: name,
+        nftCoverImageSrc: image,
+        nftURL: getNFTURL(classId),
+        unsubscribeLink,
+        language: convertLanguageCodeForEmailTemplate(language),
+      });
+      // eslint-disable-next-line no-await-in-loop
+      await sendEmail({
+        email,
+        subject,
+        html: body,
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  }
+}
+
+export async function handlePurchaseEvent(message, data) {
+  const {
+    // txHash,
+    // iscnId,
+    classId,
+    nftId,
+    nftPrice,
+    buyerWallet,
+    buyerMemo,
+    sellerWallet,
+  } = data;
+  if (!sellerWallet) {
+    throw new Error(
+      `sellerWallet is not defined for message ${message.messageId}`
+    );
+  }
+
+  const { name, image } = await fetchNFTMetadata(classId);
+
+  const userDoc = await walletUserCollection.doc(sellerWallet).get();
+  const userDocData = userDoc.data();
+  if (!userDocData) return;
+
+  const {
+    email,
+    notification: { purchasePrice = 0 } = {},
+    locale = 'en',
+  } = userDocData;
+
+  if (email && purchasePrice >= 0 && nftPrice >= purchasePrice) {
+    try {
+      const [
+        {
+          likerId: buyerLikerId,
+          avatar: buyerAvatarSrc,
+          displayName: buyerDisplayName,
+          isSubscribedCivicLiker: buyerIsCivicLiker,
+        },
+        { displayName: sellerDisplayName },
+        // eslint-disable-next-line no-await-in-loop
+      ] = await Promise.all([
+        fetchLikerInfoByWallet(buyerWallet),
+        fetchLikerInfoByWallet(sellerWallet),
+      ]);
+
+      const { subject, body } = getNFTNotificationPurchaseTemplate({
+        followerDisplayName: sellerDisplayName,
+        unsubscribeLink: TX_UNSUBSCRIBE_LINK,
+        buyerLikerId,
+        buyerDisplayName,
+        buyerAvatarSrc,
+        buyerIsCivicLiker,
+        priceInLIKE: nftPrice,
+        message: buyerMemo,
+        nftTitle: name,
+        nftCoverImageSrc: image,
+        nftURL: getNFTURL(classId, nftId),
+        language: convertLanguageCodeForEmailTemplate(locale),
+      });
+
+      // eslint-disable-next-line no-await-in-loop
+      await sendEmail({
+        email,
+        subject,
+        html: body,
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  }
+}
+
+export async function handleTransferEvent(message, data) {
+  const {
+    classId,
+    // iscnId,
+    nftId,
+    // txHash,
+    // timestamp,
+    fromAddress,
+    toAddress,
+    memo,
+  } = data;
+  if (!toAddress) {
+    throw new Error(
+      `toAddress is not defined for message ${message.messageId}`
+    );
+  }
+
+  const { name, image } = await fetchNFTMetadata(classId);
+
+  const userDoc = await walletUserCollection.doc(toAddress).get();
+  const userDocData = userDoc.data();
+  if (!userDocData) return;
+
+  const {
+    email,
+    notification: { transfer = false } = {},
```

Comment:
transfer is default to `true` in GET api, but `false` here, potential inconsistency? Maybe should be `true` here also?